src/main/java/com/conveyal/datatools/manager/models/Deployment.java

```diff
@@ -396,6 +400,32 @@ public String generateBuildConfigAsString() {
         return writeToString(this.parentProject().buildConfig);
     }
 
+    /**
+     * Check if there are active jobs to fetch or process a new version for all feed sources. This is helpful in the
+     * context of auto-deploying to OTP to determine if a given feed is the last to process a newly fetched version
+     * (we only want the deployment to occur if all fetches have completed).
+     * TODO: what if some of the feed fetches loaded new GTFS data that had critical errors? Should we not
+     *  auto-deploy?
+     */
+    public boolean hasFeedFetchesInProgress() {
+        // Collect this deployment's feed source IDs (from feed versions) to check the active jobs against.
+        Set<String> feedSourceIds = retrieveFullFeedVersions().stream()
+            .map(version -> version.feedSourceId)
+            .collect(Collectors.toSet());
+        // If there are any active fetch/process feed jobs for one of the feed source IDs, return true.
+        return StatusController.filterActiveJobs(StatusController.getAllJobs()).stream().anyMatch(job -> {
+            if (job instanceof FetchSingleFeedJob) {
+                FetchSingleFeedJob fetchJob = (FetchSingleFeedJob) job;
+                if (feedSourceIds.contains(fetchJob.feedSourceId)) return true;
+            }
+            if (job instanceof ProcessSingleFeedJob) {
+                ProcessSingleFeedJob processSingleFeedJob = (ProcessSingleFeedJob) job;
+                if (feedSourceIds.contains(processSingleFeedJob.getFeedSourceId())) return true;
+            }
+            return false;
```

Comment:
There should also be checks for any other job that could result in a ProcessSingleFeedJob being created. Jobs that seem to result in this would be as follows: `CreateFeedVersionFromSnapshotJob`, `MergeFeedsJob` and `FetchProjectFeedsJob`.

Reply:
@evansiroky I've added the first two. `FetchProjectFeedsJob` kicks off many `FetchSingleFeedJob`s which is already covered.